### PR TITLE
chore(tracing): update dd-trace-cpp to 0.1.11

### LINKED ISSUE #30552: Newer release available `com_github_datadog_dd_trace_cpp`: v0.1.11 (current: v0.1.8)
Package Name: com_github_datadog_dd_trace_cpp@0.1.8
Current Version: v0.1.8@2023-03-31
Available Version: v0.1.11@2023-10-26
Upstream releases: https://github.com/DataDog/dd-trace-cpp/releases

### PR DESCRIPTION
**Pull request by sebbrandt87**
Fix #30552

Commit Message:
chore(tracing): update dd-trace-cpp to 0.1.11

Additional Description:
- include new propagation style b3multi compatibility

see https://github.com/DataDog/dd-trace-cpp/blob/v0.1.11/src/datadog/propagation_style.cpp#L39-L49

Signed-off-by: Brandt, Sebastian (ITO-PC) <sebastian.brandt@tchibo.de>

Risk Level: Low
Testing: through implemented testing already present
Docs Changes: N/A
Release Notes: Yes
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
@sebbrandt87 there are a large number of changes here that seem unrelated to the PR

my guess is that you ran some of the tooling outside of our build container and with incorrect versions

**Comment by sebbrandt87** [comment]
> @sebbrandt87 there are a large number of changes here that seem unrelated to the PR
> 
> my guess is that you ran some of the tooling outside of our build container and with incorrect versions

indeed, let me recheck and recommit, sorry for the inconvenience

**Review by phlax — DISMISSED** [review]
lgtm, pending CI, thanks @sebbrandt87

**Review by phlax — DISMISSED** [review]

**Comment by phlax** [comment]
```console
In file included from bazel-out/aarch64-opt/bin/external/com_github_datadog_dd_trace_cpp/_virtual_includes/dd_trace_cpp/datadog/tracer.h:15:
bazel-out/aarch64-opt/bin/external/com_github_datadog_dd_trace_cpp/_virtual_includes/dd_trace_cpp/datadog/expected.h:108:9: error: object of type 'std::variant<Span, Error>' cannot be assigned because its copy assignment operator is implicitly deleted
  data_ = std::forward<Other>(other);
        ^
source/extensions/tracers/datadog/tracer.cc:109:16: note: in instantiation of function template specialization 'datadog::tracing::Expected<datadog::tracing::Span>::operator=<datadog::tracing::Span>' requested here
    maybe_span = tracer.create_span(span_config);
               ^
/opt/llvm/bin/../include/c++/v1/variant:1362:12: note: explicitly defaulted function was implicitly deleted here
  variant& operator=(const variant&) = default;
           ^
/opt/llvm/bin/../include/c++/v1/variant:1262:7: note: copy assignment operator of 'variant<datadog::tracing::Span, datadog::tracing::Error>' is implicitly deleted because base class '__sfinae_assign_base<__all<(is_copy_constructible_v<Span> && is_copy_assignable_v<Span>), (is_copy_constructible_v<Error> && is_copy_assignable_v<Error>)>::value, __all<(is_move_constructible_v<Span> && is_move_assignable_v<Span>), (is_move_constructible_v<Error> && is_move_assignable_v<Error>)>::value>' has a deleted copy assignment operator
      private __sfinae_assign_base<
      ^
/opt/llvm/bin/../include/c++/v1/__tuple:528:25: note: 'operator=' has been explicitly marked deleted here
  __sfinae_assign_base& operator=(__sfinae_assign_base const&) = delete;
                        ^
```

https://dev.azure.com/cncf/envoy/_build/results?buildId=147800&view=logs&j=767be981-567e-57d8-68c3-2140ede0a0bd&t=8a34f9ed-4946-5fcf-7cd6-99c93070884d&l=632

**Comment by phlax** [comment]
@sebbrandt87 i think the above fail looks real

**Comment by sebbrandt87** [comment]
Looks like, had no additional time to look into fixing it. If you have a quick clou how to solve it, let me know.

**Comment by adisuissa** [comment]
I think the reason is a change to the Span class in [this commit](https://github.com/DataDog/dd-trace-cpp/commit/9ac719cc1d4b6ea0a06313344c2ee432da8c9da5).

I'm not sure why the copy c'tor was [deleted](https://github.com/DataDog/dd-trace-cpp/pull/24/files#diff-49bd7f33ba3e1273b65b725ca56046a77cdb3f138ae06eeae71b2e5db696ebd7R79).

/wait-any

**Comment by phlax** [comment]
seems you need to merge main @sebbrandt87

File: changelogs/current.yaml
Comment (adisuissa) [inline changelogs/current.yaml:396]: Please refine the comment to clarify that this is part of an upgrade of the datadog tracing library.

**Review by adisuissa — COMMENTED** [review]
Thanks for doing the cross-repo fixes.
Left a small comment (also need to fix the conflict), but otherwise LGTM.

**Comment by wbpcode** [comment]
/wait

Waiting for main merging and comments to be resolved.

**Review by adisuissa — DISMISSED** [review]
LGTM, thanks!

**Comment by sebbrandt87** [comment]
I try my best with spare time I will find. Any advice or helping hand in solving much appreciated. 🤝

**Comment by adisuissa** [comment]
> I try my best with spare time I will find. Any advice or helping hand in solving much appreciated. 🤝

I don't have cycles to address this. The cause for the issue is described in a prior [comment](https://github.com/envoyproxy/envoy/pull/29354#issuecomment-1715771622).
This needs either a patch when importing the dependency, or updating the external dependency and importing the newer version (preferred).

/wait

**Comment by sebbrandt87** [comment]
> > I try my best with spare time I will find. Any advice or helping hand in solving much appreciated. 🤝
> 
> I don't have cycles to address this. The cause for the issue is described in a prior [comment](https://github.com/envoyproxy/envoy/pull/29354#issuecomment-1715771622). This needs either a patch when importing the dependency, or updating the external dependency and importing the newer version (preferred).
> 
> /wait

Okay.
https://github.com/DataDog/dd-trace-cpp/pull/24/commits/691b1904c9ddb0b03872c66f2bc53a1c5bc2d9ee seems to be forbidden to assign to the Span directly.

**Comment by phlax** [comment]
0.1.11 is available https://github.com/envoyproxy/envoy/issues/30552

**Comment by adisuissa** [comment]
Still same error as before, waiting for a solution to the underlying issue.
/wait

**Comment by phlax** [comment]
@sebbrandt87 i think this has been superseded we are now up-to-date for this dep - so im going to close - thanks for working on it

**Pull request closed without merging**
